cpython/patches/0009-Skip-wasm-gc-on-iOS-Safari-where-it-s-broken.patch

```diff
@@ -0,0 +1,59 @@
+From c627686651a46118b2a653fa61673dfc158442d4 Mon Sep 17 00:00:00 2001
+From: =?UTF-8?q?=C5=81ukasz=20Langa?= <lukasz@langa.pl>
+Date: Fri, 21 Feb 2025 19:24:41 +0100
+Subject: [PATCH 9/9] Skip wasm-gc on iOS Safari where it's broken
+
+As of iOS 18.3.1, enabling wasm-gc is making the interpreter fail to load.
+Downstream pyodide issue: pyodide/pyodide#5428.
+
+macOS Safari 18.3 does not surface the issue.
+
+Confirmed on device that disabling this restores interpreter function.
+---
+ Python/emscripten_trampoline.c | 16 +++++++++++++---
+ 1 file changed, 13 insertions(+), 3 deletions(-)
+
+diff --git a/Python/emscripten_trampoline.c b/Python/emscripten_trampoline.c
+index e78a94e5e99..ff57f9e91d8 100644
+--- a/Python/emscripten_trampoline.c
++++ b/Python/emscripten_trampoline.c
+@@ -79,7 +79,13 @@ EM_JS(CountArgsFunc, _PyEM_GetCountArgsPtr, (), {
+ //         i32.const -1
+ //     )
+ // )
+-addOnPreRun(() => {
++
++function getPyEMCountArgsPtr() {
++    let isIOS = globalThis.navigator && /iPad|iPhone|iPod/.test(navigator.platform);
```

Comment:
ouch ... was this *really* an *iOS* only thing? I understand the ugly workaround but I wonder how comes *iOS* would branch that much from *iPad* and I had a honest little laugh at reading *iPod* in there 😁 (nostalgic personal hysterical 
laugh I believe)

Reply:
So - it's been reported on iOS 18.3.1; iPads running it (or... in theory, iPods, but I don't know if there's any actually still supported that could run iPod) should also be affected, but I can't say I've tried personally. 

And I completely agree that UA sniffing isn't an ideal solution - but the perfect is the enemy of the good until we're able to determine a better underlying fix based on sniffing specific behavior. At this point, all we know is "iOS 18.3.1(+?) bad", and we have a fallback.